script.sublissimo/resources/lib/menu_items/syncing.py

```diff
@@ -44,30 +44,18 @@ def check_player_instances(subtitle=None):
     if subtitle:
         subtitle.delete_temp_file()
 
-def experiment(typeofinstance, subtitle):
-    current_player = xbmc.Player()
-    if current_player.isPlayingVideo():
-        if current_player.getPlayingFile() == subtitle.videofilename:
-            player = PlayerInstance().request(typeofinstance)
-            player.add(subtitle)
-            player.start()
-            xbmc.Monitor().waitForAbort(1)
-    else:
-        return False
-
 def create_player(typeofinstance, subtitle):
-    if not experiment(typeofinstance, subtitle):
-        player = PlayerInstance().request(typeofinstance)
-        location = retrieve_video(subtitle)
-        player.add(subtitle)
-        player.play(location)
-        player.start()
-        # -----Possibly needed if video is unavailable-----
-        # xbmc.sleep(1000)
-        # if player.isPlaying() == False:
-        #     show_dialog(subtitle)
-        # -------------------------------------------------
-        xbmc.Monitor().waitForAbort(1)
+    player = PlayerInstance().request(typeofinstance)
+    location = retrieve_video(subtitle)
+    player.add(subtitle)
+    player.play(location)
+    player.start()
+    # -----Possibly needed if video is unavailable-----
+    # xbmc.sleep(1000)
+    # if player.isPlaying() == False:
+    #     show_dialog(subtitle)
+    # -------------------------------------------------
+    xbmc.Monitor().waitForAbort(1000)
```

Comment:
this is 1000 seconds of pause. Why do you need such a long pause?
Most likely a clear sign something is wrong with your logic... Do you want to keep something alive while the player is active or so?

Reply:
For the sync option the user needs time to select a begin and an endpoint to sync the subtitle with. In the older versions of this script, I always used xbmc.sleep(1000). Most users (well I at least) select these two point within 5 minutes, and then exit the player. Without it, the menu's simply don't appear.

Reply:
Do have access to the maximum time interval? Something like this should work:

```
while not monitor.abortRequested():
   if something:
       break
   monitor.waitForAbort(1)
```

Reply:
Well that seems to work! Is this solution okay?